test/integration/node-specific/bson-options/promote_values.test.ts

```diff
@@ -1,209 +1,172 @@
-        int: 10,
-        double: 2.2222,
-        array: [[Long.fromNumber(10)]]
-      });
-
-      const doc = await db
-        .collection('shouldCorrectlyHonorPromoteValues')
-        .find({}, { promoteValues: false })
-        .next();
-      expect(doc.doc).to.deep.equal(Long.fromNumber(10));
-      expect(doc.int).to.deep.equal(new Int32(10));
-      expect(doc.double).to.deep.equal(new Double(2.2222));
-
-      await client.close();
-    }
+  it('should correctly honor promoteValues at aggregate level', async function () {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+    await client.connect();
+    const db = client.db(configuration.db);
+    await db.collection('shouldCorrectlyHonorPromoteValues2').insertOne({
+      doc: Long.fromNumber(10),
+      int: 10,
+      double: 2.2222,
+      array: [[Long.fromNumber(10)]]
+    });
+    const doc = await db
+      .collection('shouldCorrectlyHonorPromoteValues2')
+      .aggregate([{ $match: {} }], { promoteValues: false })
+      .next();
+    expect(doc.doc).to.deep.equal(Long.fromNumber(10));
+    expect(doc.int).to.deep.equal(new Int32(10));
+    expect(doc.double).to.deep.equal(new Double(2.2222));
+
+    await client.close();
   });
 
-  it('should correctly honor promoteValues at aggregate level', {
-    // Add a tag that our runner can trigger on
-    // in this case we are setting that node needs to be higher than 0.10.X to run
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    test: async function () {
-      const configuration = this.configuration;
-      const client = configuration.newClient();
-      await client.connect();
-      const db = client.db(configuration.db);
-      await db.collection('shouldCorrectlyHonorPromoteValues2').insertOne({
-        doc: Long.fromNumber(10),
-        int: 10,
-        double: 2.2222,
-        array: [[Long.fromNumber(10)]]
-      });
-      const doc = await db
-        .collection('shouldCorrectlyHonorPromoteValues2')
-        .aggregate([{ $match: {} }], { promoteValues: false })
-        .next();
-      expect(doc.doc, Long.fromNumber(10));
-      expect(doc.int, new Int32(10));
-      expect(doc.double, new Double(2.2222));
-
-      await client.close();
+  it('Should correctly promoteValues when calling getMore on queries', async function () {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+    const docs = new Array(150).fill(0).map(function (_, i) {
+      return {
+        _id: 'needle_' + i,
+        is_even: i % 2,
+        long: Long.fromString('1234567890'),
+        double: 0.23456,
+        int: 1234
+      };
+    });
+
+    const db = client.db(configuration.db);
+
+    await db.collection<{ _id: string }>('haystack').insertMany(docs);
+
+    const stream = db
+      .collection('haystack')
+      .find({}, { limit: 102, promoteValues: false })
```

Comment:
```suggestion
      .find({}, { batchSize: 50, promoteValues: false })
```

Just to explicitly force multiple batches